# AirBnB NY Locations Data Case Study

In this final project, you task will be to take the data provided and find evidance to answer the following questions.

1. Which hosts are the busiest and why?
2. How many neighborhood groups are available and which shows up the most?
3. Are private rooms the most popular in manhattan?
4. Which hosts are the busiest and based on their reviews?
5. Which neighorhood group has the highest average price?
6. Which neighborhood group has the highest total price?
7. Which top 5 hosts have the highest total price?
8. Who currently has no (zero) availability with a review count of 100 or more?
9. What host has the highest total of prices and where are they located?
10. When did Danielle from Queens last receive a review?

You will be given **4 hours** to complete this assignment. 
**Be Advised** I will go dark for this intire assignment time period. That said, any questions that you would like to ask about the data, or the project **MUST** be asked before the time starts. Once the time has started, I can no longer give information.

This is to similate what you will face when you are out in the wild. 

Happy Coding!

In [1]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt

In [2]:
air_bnb = pd.read_csv('AB_NYC_2019.csv')
air_bnb.head()

id                                              name  host_id  \
0  2539                Clean & quiet apt home by the park     2787   
1  2595                             Skylit Midtown Castle     2845   
2  3647               THE VILLAGE OF HARLEM....NEW YORK !     4632   
3  3831                   Cozy Entire Floor of Brownstone     4869   
4  5022  Entire Apt: Spacious Studio/Loft by central park     7192   

     host_name neighbourhood_group neighbourhood  latitude  longitude  \
0         John            Brooklyn    Kensington  40.64749  -73.97237   
1     Jennifer           Manhattan       Midtown  40.75362  -73.98377   
2    Elisabeth           Manhattan        Harlem  40.80902  -73.94190   
3  LisaRoxanne            Brooklyn  Clinton Hill  40.68514  -73.95976   
4        Laura           Manhattan   East Harlem  40.79851  -73.94399   

         room_type  price  minimum_nights  number_of_reviews last_review  \
0     Private room    149               1                  9  2018-10-19   
1  Entire home/apt    225               1                 45  2019-05-21   
2     Private room    150               3                  0         NaN   
3  Entire home/apt     89               1                270  2019-07-05   
4  Entire home/apt     80              10                  9  2018-11-19   

   reviews_per_month  calculated_host_listings_count  availability_365  
0               0.21                               6               365  
1               0.38                               2               355  
2                NaN                               1               365  
3               4.64                               1               194  
4               0.10                               1                 0

In [3]:
# How many neighborhood groups are available and which shows up the most?

neighborhood_groups_available = air_bnb['neighbourhood_group'].nunique()
neighborhood_group_that_shows_up_most = air_bnb['neighbourhood_group'].max()
print(f'There are {neighborhood_groups_available} neighborhood groups available.')
print(f'{neighborhood_group_that_shows_up_most} is the neighborhood group that shows up the most.')

There are 5 neighborhood groups available.
Staten Island is the neighborhood group that shows up the most.


In [4]:
# Are private rooms the most popular in manhattan?
most_popular_room = air_bnb['room_type'].max()
print(most_popular_room == 'Private room')
print(f"{most_popular_room}'s are the most popular type of room.")

False
Shared room's are the most popular type of room.


In [5]:
# Which hosts are the busiest and based on their reviews?
reviews_per_user = dict()
for row, col in air_bnb.iterrows():
    if col['host_id'] not in reviews_per_user:
        reviews_per_user[col['host_id']] = col['number_of_reviews']
    else: 
        reviews_per_user[col['host_id']] += col['number_of_reviews']
host_with_max_reviews = max(reviews_per_user, key = reviews_per_user.get)
host_with_max_reviews_df = air_bnb.loc[air_bnb['host_id'] == host_with_max_reviews]
print(f'The host with host_id {host_with_max_reviews} has the most reviews overall.')
host_with_max_reviews_df

The host with host_id 37312959 has the most reviews overall.


id                                    name   host_id host_name  \
9323    7125107  Room steps away from LaGuardia airport  37312959      Maya   
10623   8168619       Steps away from Laguardia airport  37312959      Maya   
13612  10186192     Only Steps away from LaGuardia arpt  37312959      Maya   
14699  11618854  Walking distance  to LaGuardia airport  37312959      Maya   
18045  14133414    Space to rest near LaGuardia Airport  37312959      Maya   

      neighbourhood_group  neighbourhood  latitude  longitude     room_type  \
9323               Queens  East Elmhurst  40.77005  -73.87691  Private room   
10623              Queens  East Elmhurst  40.77006  -73.87683  Private room   
13612              Queens  East Elmhurst  40.77026  -73.87561  Private room   
14699              Queens  East Elmhurst  40.77107  -73.87559  Private room   
18045              Queens  East Elmhurst  40.77009  -73.87708  Private room   

       price  minimum_nights  number_of_reviews last_review  \
9323      45               1                448  2019-07-07   
10623     46               1                543  2019-07-01   
13612     45               1                459  2019-07-07   
14699     45               1                412  2019-07-01   
18045     32               1                411  2019-07-04   

       reviews_per_month  calculated_host_listings_count  availability_365  
9323                9.63                               5               166  
10623              11.59                               5               163  
13612              10.72                               5               175  
14699              10.19                               5               159  
18045              11.40                               5               161

In [6]:
reviews_per_user_per_month = dict()
for row, col in air_bnb.iterrows():
    if col['host_id'] not in reviews_per_user_per_month:
        reviews_per_user_per_month[col['host_id']] = col['reviews_per_month']
    else: 
        reviews_per_user_per_month[col['host_id']] = (col['reviews_per_month']+reviews_per_user_per_month[col['host_id']])/2
host_with_max_reviews_per_month = max(reviews_per_user_per_month, key = reviews_per_user_per_month.get)
host_with_max_reviews_per_month_df = air_bnb.loc[air_bnb['host_id'] == host_with_max_reviews_per_month]
print(f'The host with host_id {host_with_max_reviews_per_month} gets the most reviews per month.')
host_with_max_reviews_per_month_df

The host with host_id 228415932 gets the most reviews per month.


id                                name    host_id host_name  \
38870  30423106  Lou's Palace-So much for so little  228415932    Louann   

      neighbourhood_group neighbourhood  latitude  longitude     room_type  \
38870              Queens      Rosedale  40.65417  -73.74158  Private room   

       price  minimum_nights  number_of_reviews last_review  \
38870     45               1                 37  2019-07-08   

       reviews_per_month  calculated_host_listings_count  availability_365  
38870              20.94                               1               134

In [7]:
print(f'{host_with_max_reviews_df.loc[9323, "host_name"]} is the busiest host according to the tally of total reviews.')
print(f'{host_with_max_reviews_per_month_df.loc[38870, "host_name"]} is the busiest host accourding to the tally of reviews per month.')


Maya is the busiest host according to the tally of total reviews.
Louann is the busiest host accourding to the tally of reviews per month.


In [8]:
#Which neighorhood group has the highest average price?
avg_price_by_hood_group = dict()
for row, col in air_bnb.iterrows():
    if col['neighbourhood_group'] not in avg_price_by_hood_group:
        avg_price_by_hood_group[col['neighbourhood_group']] = col['price']
    else: 
        avg_price_by_hood_group[col['neighbourhood_group']] = (col['price']+avg_price_by_hood_group[col['neighbourhood_group']])/2
        
neighborhood_with_highest_avg_price = max(avg_price_by_hood_group, key=avg_price_by_hood_group.get)
neighborhood_with_highest_avg_price_df = air_bnb.loc[air_bnb['neighbourhood_group'] == neighborhood_with_highest_avg_price]
highest_avg_price = neighborhood_with_highest_avg_price_df['price'].mean()
print(f'The neighborhood group that has the highest average price at ${highest_avg_price:.2f} is {neighborhood_with_highest_avg_price}')
neighborhood_with_highest_avg_price_df
        

The neighborhood group that has the highest average price at $114.81 is Staten Island


id                                               name    host_id  \
169       42882                          New York room with a view     185978   
249       62452                     A SpeciaL!! Private Room in NY     303939   
250       62461                        B NYC Staten Alternative...     303939   
251       62787                        C Private Room By The Ferry     303939   
256       63320                 D Private Che@p Room 2 Explore NYC     303939   
...         ...                                                ...        ...   
48441  36267546                              A Home Away From Home  233485864   
48446  36270376     Beautiful home on Staten Island, walk to ferry     444904   
48533  36310353  Staten Island Resort minutes from ferry! Amazi...   28586423   
48649  36355110     PRIVATE ROOM IN STATEN ISLAND -FOR LADIES ONLY  201187671   
48799  36438336                                    Seas The Moment  211644523   

                  host_name neighbourhood_group   neighbourhood  latitude  \
169    Newyorkroomwithaview       Staten Island      St. George  40.64524   
249                Lissette       Staten Island   Tompkinsville  40.63536   
250                Lissette       Staten Island   Tompkinsville  40.63627   
251                Lissette       Staten Island   Tompkinsville  40.63518   
256                Lissette       Staten Island   Tompkinsville  40.63481   
...                     ...                 ...             ...       ...   
48441               Jeannie       Staten Island  New Dorp Beach  40.56506   
48446                Janine       Staten Island      St. George  40.64005   
48533              Danielle       Staten Island       Stapleton  40.62878   
48649                 Tuana       Staten Island     Shore Acres  40.61283   
48799                   Ben       Staten Island     Great Kills  40.54179   

       longitude        room_type  price  minimum_nights  number_of_reviews  \
169    -74.08088     Private room     70               2                166   
249    -74.08537     Private room     36               2                193   
250    -74.08543     Private room     37               2                147   
251    -74.08546     Private room     37               2                177   
256    -74.08519     Private room     36               2                333   
...          ...              ...    ...             ...                ...   
48441  -74.10240     Private room     40               1                  0   
48446  -74.07987  Entire home/apt    100             365                  0   
48533  -74.07301  Entire home/apt    450               5                  0   
48649  -74.06625     Private room     54               4                  0   
48799  -74.14275     Private room    235               1                  1   

      last_review  reviews_per_month  calculated_host_listings_count  \
169    2019-06-13               1.66                               1   
249    2019-06-25               1.85                               6   
250    2019-06-10               1.44                               6   
251    2019-07-02               1.71                               6   
256    2019-07-02               3.19                               6   
...           ...                ...                             ...   
48441         NaN                NaN                               2   
48446         NaN                NaN                               1   
48533         NaN                NaN                               1   
48649         NaN                NaN                               1   
48799  2019-07-07               1.00                               1   

       availability_365  
169                 312  
249                 360  
250                   0  
251                 320  
256                 340  
...                 ...  
48441               341  
48446               342  
48533                88  
48649                89  
48799          

In [9]:
# Which neighbor hood group has the highest total price?
total_price_by_hood_group = dict()
for row, col in air_bnb.iterrows():
    if col['neighbourhood_group'] not in total_price_by_hood_group:
        total_price_by_hood_group[col['neighbourhood_group']] = col['price']
    else: 
        total_price_by_hood_group[col['neighbourhood_group']] += col['price']
        
neighborhood_with_highest_total_price = max(total_price_by_hood_group, key=total_price_by_hood_group.get)
neighborhood_with_highest_total_price_df = air_bnb.loc[air_bnb['neighbourhood_group'] == neighborhood_with_highest_total_price]
highest_total_price = neighborhood_with_highest_total_price_df["price"].sum()
print(f'The neighborhood group that has the highest total price at ${highest_total_price} is {neighborhood_with_highest_total_price}')
neighborhood_with_highest_total_price_df


The neighborhood group that has the highest total price at $4264527 is Manhattan


id                                               name    host_id  \
1          2595                              Skylit Midtown Castle       2845   
2          3647                THE VILLAGE OF HARLEM....NEW YORK !       4632   
4          5022   Entire Apt: Spacious Studio/Loft by central park       7192   
5          5099          Large Cozy 1 BR Apartment In Midtown East       7322   
7          5178                   Large Furnished Room Near B'way        8967   
...         ...                                                ...        ...   
48886  36483010                    Comfy 1 Bedroom in Midtown East  274311461   
48888  36484087  Spacious Room w/ Private Rooftop, Central loca...  274321313   
48892  36485431            Sunny Studio at Historical Neighborhood   23492952   
48893  36485609               43rd St. Time Square-cozy single bed   30985759   
48894  36487245  Trendy duplex in the very heart of Hell's Kitchen   68119814   

           host_name neighbourhood_group   neighbourhood  latitude  longitude  \
1           Jennifer           Manhattan         Midtown  40.75362  -73.98377   
2          Elisabeth           Manhattan          Harlem  40.80902  -73.94190   
4              Laura           Manhattan     East Harlem  40.79851  -73.94399   
5              Chris           Manhattan     Murray Hill  40.74767  -73.97500   
7           Shunichi           Manhattan  Hell's Kitchen  40.76489  -73.98493   
...              ...                 ...             ...       ...        ...   
48886          Scott           Manhattan         Midtown  40.75561  -73.96723   
48888            Kat           Manhattan  Hell's Kitchen  40.76392  -73.99183   
48892  Ilgar & Aysel           Manhattan          Harlem  40.81475  -73.94867   
48893            Taz           Manhattan  Hell's Kitchen  40.75751  -73.99112   
48894     Christophe           Manhattan  Hell's Kitchen  40.76404  -73.98933   

             room_type  price  minimum_nights  number_of_reviews last_review  \
1      Entire home/apt    225               1                 45  2019-05-21   
2         Private room    150               3                  0         NaN   
4      Entire home/apt     80              10                  9  2018-11-19   
5      Entire home/apt    200               3                 74  2019-06-22   
7         Private room     79               2                430  2019-06-24   
...                ...    ...             ...                ...         ...   
48886  Entire home/apt    200               6                  0         NaN   
48888     Private room    125               4                  0         NaN   
48892  Entire home/apt    115              10                  0         NaN   
48893      Shared room     55               1                  0         NaN   
48894     Private room     90               7                  0         NaN   

       reviews_per_month  calculated_host_listings_count  availability_365  
1                   0.38                               2               355  
2                    NaN                               1               365  
4                   0.10                               1                 0  
5                   0.59                               1               129  
7                   3.47                               1               220  
...                  ...                             ...               ...  
48886                NaN                               1               176  
48888                NaN                               1                31  
48892                NaN                               1                27  
48893                NaN                               6                 2  
48894                NaN                               1                23  

[21661 rows x 16 columns]

In [14]:
#Which top 5 hosts have the highest total price?
air_bnb.sort_values(by=['price'], ascending=False).head()

id                                               name   host_id  \
9151    7003697                Furnished room in Astoria apartment  20582832   
17692  13894339    Luxury 1 bedroom apt. -stunning Manhattan views   5143901   
29238  22436899                                1-BR Lincoln Center  72390391   
40433  31340283  2br - The Heart of NYC: Manhattans Lower East ...   4382127   
12342   9528920                Quiet, Clean, Lit @ LES & Chinatown   3906464   

      host_name neighbourhood_group    neighbourhood  latitude  longitude  \
9151   Kathrine              Queens          Astoria  40.76810  -73.91651   
17692      Erin            Brooklyn       Greenpoint  40.73260  -73.95739   
29238    Jelena           Manhattan  Upper West Side  40.77213  -73.98665   
40433      Matt           Manhattan  Lower East Side  40.71980  -73.98566   
12342       Amy           Manhattan  Lower East Side  40.71355  -73.98507   

             room_type  price  minimum_nights  number_of_reviews last_review  \
9151      Private room  10000             100                  2  2016-02-13   
17692  Entire home/apt  10000               5                  5  2017-07-27   
29238  Entire home/apt  10000              30                  0         NaN   
40433  Entire home/apt   9999              30                  0         NaN   
12342     Private room   9999              99                  6  2016-01-01   

       reviews_per_month  calculated_host_listings_count  availability_365  
9151                0.04                               1                 0  
17692               0.16                               1                 0  
29238                NaN                               1                83  
40433                NaN                               1               365  
12342               0.14                               1                83

In [15]:
# Who currently has no (zero) availability with a review count of 100 or more?
ans = air_bnb.loc[(air_bnb['availability_365']==0) & (air_bnb['number_of_reviews']>=100)]
print(ans.host_name.unique())
ans

['MaryEllen' 'Christiana' 'Sol' 'Coral' 'Doug' 'Ori' 'Lissette'
 'Liz And Melissa' 'Ivy' 'Jsun' 'Wanda' 'Ben' 'S' 'Adrienne' 'Lydia'
 'Karin' 'Elle' 'James' 'Jon' 'Liz' 'Jeanine' 'Lorena' 'Ron' 'Dragan'
 'Misty' 'Brian' 'Natalie' 'AJ And Freddy' 'Neil & Katie' 'Emily' 'Evelyn'
 'Alvaro' 'Bernard' 'Sarah' 'Karen' 'Summer' 'William' 'Andy & Friends'
 'Karece' 'Ehren' 'Nicole' 'Terri' 'Ravanna' 'Molly' 'Lane' 'Angelo'
 'DeLex' 'Michelle' 'Katarina' 'Véronique' 'Andreas' 'Caroline' 'Michael'
 'Aurea' 'Kent' 'Brendan' 'Jillian' 'Deanna' 'Jake' 'Emily And Joel'
 'George & Diana' 'Veronica' 'Masha' 'Danielle' 'Jeremy' 'Kyle' 'Stacey'
 'Sasha' 'Nick' 'Carlina' 'Taylor & Tee' 'Devin' 'Ame' 'Richard' 'Micah'
 'Elliott' '正川' 'Chao' 'Pj' 'Lou' 'Ingrid' 'Graham' 'Gurpreet  Singh'
 'Jimmy' 'Catrina' 'Long' 'Deborah' 'Hayes' 'Evan' 'Sofia' 'Antonia'
 'Margarita' 'Abraham' 'Alex' 'Qiyao' 'Cedrick' 'Greg' 'Chelsea' 'Kc'
 'Edward' 'Lasata' 'Krysta' 'Maeve' 'Erin' 'Trevor' 'Giovanni' 'Anthony'
 'Chadanut

id                                       name    host_id  \
8          5203         Cozy Clean Guest Room - Family Apt       7490   
94        20913         Charming 1 bed GR8 WBurg LOCATION!      79402   
132       30031           NYC artists’ loft with roof deck     129352   
174       44221             Financial District Luxury Loft     193722   
180       45556      Fort Greene, Brooklyn: Center Bedroom      67778   
...         ...                                        ...        ...   
29581  22705516         The Quietest Block in Manhattan :)  127740507   
30461  23574142                          queens get away!!  176185168   
31250  24267706  entire sunshine of the spotless mind room   21074914   
32670  25719044                COZY Room for Female Guests   40119874   
35014  27759146     Cozy corner near Empire State Building  209549523   

        host_name neighbourhood_group              neighbourhood  latitude  \
8       MaryEllen           Manhattan            Upper West Side  40.80178   
94     Christiana            Brooklyn               Williamsburg  40.70984   
132           Sol            Brooklyn                 Greenpoint  40.73494   
174         Coral           Manhattan         Financial District  40.70666   
180          Doug            Brooklyn                Fort Greene  40.68863   
...           ...                 ...                        ...       ...   
29581    Kathleen           Manhattan                     Harlem  40.83102   
30461       Janet              Queens                  Laurelton  40.68209   
31250      Albert            Brooklyn         Bedford-Stuyvesant  40.68234   
32670    Stephany            Brooklyn  Prospect-Lefferts Gardens  40.66242   
35014     Mariluz           Manhattan                    Midtown  40.74858   

       longitude        room_type  price  minimum_nights  number_of_reviews  \
8      -73.96723     Private room     79               2                118   
94     -73.95775  Entire home/apt    100               5                168   
132    -73.95030     Private room     50               3                193   
174    -74.01374  Entire home/apt    196               3                114   
180    -73.97691     Private room     65               2                206   
...          ...              ...    ...             ...                ...   
29581  -73.94181     Private room     65               2                103   
30461  -73.73662     Private room     65               1                119   
31250  -73.91318     Private room     49               1                102   
32670  -73.94417     Private room     48               1                131   
35014  -73.98341      Shared room     62               1                112   

      last_review  reviews_per_month  calculated_host_listings_count  \
8      2017-07-21               0.99                               1   
94     2018-07-22               1.57                               1   
132    2019-05-20               1.86                               1   
174    2019-06-20               1.06                               1   
180    2019-06-30               1.92                               2   
...           ...                ...                             ...   
29581  2019-07-07               5.89                               2   
30461  2018-12-24               7.79                               1   
31250  2019-07-05               6.73                               3   
32670  2019-05-31               9.97                               2   
35014  2019-06-13              10.77                               3   

       availability_365  
8                     0  
94                    0  
132                   0  
174                   0  
180                   0  
...                 ...  
29581                 0  
30461                 0  
31250                 0  
32670                 0  
35014                 0  

[162 rows x 16 columns]

In [16]:
# What host has the highest total of prices and where are they located?
total_price_by_host = dict()
for row, col in air_bnb.iterrows():
    if col['host_name'] not in total_price_by_host:
        total_price_by_host[col['host_name']] = col['price']
    else: 
        total_price_by_host[col['host_name']] += col['price']
        
host_with_highest_total_prices = max(total_price_by_host, key=total_price_by_host.get)
host_with_highest_total_prices_df = air_bnb.loc[air_bnb['host_name'] == host_with_highest_total_prices]
highest_total_prices = host_with_highest_total_prices_df["price"].sum()
host_with_highest_total_prices_location = host_with_highest_total_prices_df.loc[38293, 'neighbourhood_group']
print(f'The host that has the highest total prices at ${highest_total_prices} is {host_with_highest_total_prices} who is located in {host_with_highest_total_prices_location}')
host_with_highest_total_prices_df

The host that has the highest total prices at $82795 is Sonder (NYC) who is located in Manhattan


id                                               name    host_id  \
38293  30181691      Sonder | 180 Water | Incredible 2BR + Rooftop  219517861   
38294  30181945         Sonder | 180 Water | Premier 1BR + Rooftop  219517861   
38588  30347708        Sonder | 180 Water | Charming 1BR + Rooftop  219517861   
39769  30937590            Sonder | The Nash | Artsy 1BR + Rooftop  219517861   
39770  30937591        Sonder | The Nash | Lovely Studio + Rooftop  219517861   
...         ...                                                ...        ...   
47691  35871510  Sonder | 116 John | Vibrant Studio + Fitness Room  219517861   
47692  35871511     Sonder | 116 John | Vibrant 1BR + Fitness Room  219517861   
47693  35871515         Sonder | 116 John | Stunning 1BR + Rooftop  219517861   
47814  35936418          Sonder | 116 John | Polished Studio + Gym  219517861   
47821  35937891            Sonder | 116 John | Simple Studio + Gym  219517861   

          host_name neighbourhood_group       neighbourhood  latitude  \
38293  Sonder (NYC)           Manhattan  Financial District  40.70637   
38294  Sonder (NYC)           Manhattan  Financial District  40.70771   
38588  Sonder (NYC)           Manhattan  Financial District  40.70743   
39769  Sonder (NYC)           Manhattan         Murray Hill  40.74792   
39770  Sonder (NYC)           Manhattan         Murray Hill  40.74771   
...             ...                 ...                 ...       ...   
47691  Sonder (NYC)           Manhattan  Financial District  40.70818   
47692  Sonder (NYC)           Manhattan  Financial District  40.70691   
47693  Sonder (NYC)           Manhattan  Financial District  40.70772   
47814  Sonder (NYC)           Manhattan  Financial District  40.70840   
47821  Sonder (NYC)           Manhattan  Financial District  40.70707   

       longitude        room_type  price  minimum_nights  number_of_reviews  \
38293  -74.00645  Entire home/apt    302              29                  0   
38294  -74.00641  Entire home/apt    229              29                  1   
38588  -74.00443  Entire home/apt    232              29                  1   
39769  -73.97614  Entire home/apt    262               2                  8   
39770  -73.97528  Entire home/apt    255               2                 14   
...          ...              ...    ...             ...                ...   
47691  -74.00631  Entire home/apt    135              29                  0   
47692  -74.00682  Entire home/apt    165              29                  0   
47693  -74.00673  Entire home/apt    165              29                  0   
47814  -74.00518  Entire home/apt    699              29                  0   
47821  -74.00557  Entire home/apt    699              29                  0   

      last_review  reviews_per_month  calculated_host_listings_count  \
38293         NaN                NaN                             327   
38294  2019-05-29               0.73                             327   
38588  2019-05-21               0.60                             327   
39769  2019-06-09               1.86                             327   
39770  2019-06-10               2.59                             327   
...           ...                ...                             ...   
47691         NaN                NaN                             327   
47692         NaN                NaN                             327   
47693         NaN                NaN                             327   
47814         NaN                NaN                             327   
47821         NaN                NaN                             327   

       availability_365  
38293               309  
38294               219  
38588               159  
39769                91  
39770                81  
...                 ...  
47691               339  
47692               342  
47693               347  
47814               327  
47821               341  

[327 rows x 16 columns]

In [17]:
# When did Danielle from Queens last receive a review?
danielle_from_queens_listings = air_bnb.loc[(air_bnb['host_name']=='Danielle') & (air_bnb['neighbourhood_group']=='Queens')]
danielle_from_queens_last_review = pd.to_datetime(danielle_from_queens_listings['last_review']).max()
print(f'Danielle from Queens last received a review on {danielle_from_queens_last_review}')
danielle_from_queens_listings


Danielle from Queens last received a review on 2019-07-08 00:00:00


id                                               name    host_id  \
7086    5115372  Comfy Room Family Home LGA Airport NO CLEANING...   26432133   
16349  13151075                    ASTORIA APARTMENT OUTDOOR SPACE   18051286   
20403  16276632  Cozy Room Family Home LGA Airport NO CLEANING FEE   26432133   
21517  17222454   Sun Room Family Home LGA Airport NO CLEANING FEE   26432133   
22068  17754072                Bed in Family Home Near LGA Airport   26432133   
22469  18173787  Cute Tiny Room Family Home by LGA NO CLEANING FEE   26432133   
27021  21386105  Quiet & clean 1br haven with balcony near the ...  154256662   
33861  26814763  One bedroom with full bed / 1 stop from Manhattan  201647469   

      host_name neighbourhood_group     neighbourhood  latitude  longitude  \
7086   Danielle              Queens     East Elmhurst  40.76374  -73.87103   
16349  Danielle              Queens           Astoria  40.77221  -73.92901   
20403  Danielle              Queens     East Elmhurst  40.76335  -73.87007   
21517  Danielle              Queens     East Elmhurst  40.76367  -73.87088   
22068  Danielle              Queens     East Elmhurst  40.76389  -73.87155   
22469  Danielle              Queens     East Elmhurst  40.76380  -73.87238   
27021  Danielle              Queens           Astoria  40.77134  -73.92424   
33861  Danielle              Queens  Long Island City  40.74565  -73.94699   

             room_type  price  minimum_nights  number_of_reviews last_review  \
7086      Private room     54               1                430  2019-07-03   
16349     Private room     50               1                  0         NaN   
20403     Private room     48               1                510  2019-07-06   
21517     Private room     48               1                417  2019-07-07   
22068      Shared room     38               1                224  2019-07-06   
22469     Private room     48               1                436  2019-07-08   
27021  Entire home/apt    250               3                  1  2018-01-02   
33861     Private room    108               2                 13  2019-06-20   

       reviews_per_month  calculated_host_listings_count  availability_365  
7086               13.45                               5               347  
16349                NaN                               1                 0  
20403              16.22                               5               341  
21517              14.36                               5               338  
22068               7.96                               5                80  
22469              16.03                               5               337  
27021               0.05                               1               180  
33861               1.74                               1               333

## Further Questions

1. Which host has the most listings?

In [18]:
host_with_most_listings = air_bnb['host_name'].mode()
host_with_most_listings


0    Michael
dtype: object

2. How many listings have completely open availability?

In [19]:
count_listings_with_open_availability = (air_bnb.availability_365 == 365).sum()
count_listings_with_open_availability

1295

3. What room_types have the highest review numbers?

In [20]:
reviews_per_room_type = dict()
for row, col in air_bnb.iterrows():
    if col['room_type'] not in reviews_per_room_type:
        reviews_per_room_type[col['room_type']] = col['number_of_reviews']
    else: 
        reviews_per_room_type[col['room_type']] += col['number_of_reviews']
room_type_with_max_reviews = max(reviews_per_room_type, key = reviews_per_room_type.get)
room_type_with_max_reviews_df = air_bnb.loc[air_bnb['room_type'] == room_type_with_max_reviews]
print(f"{room_type_with_max_reviews}'s have the most reviews overall.\n")
print(reviews_per_room_type)
room_type_with_max_reviews_df
 

Entire home/apt's have the most reviews overall.

{'Private room': 538346, 'Entire home/apt': 580403, 'Shared room': 19256}


id                                               name    host_id  \
1          2595                              Skylit Midtown Castle       2845   
3          3831                    Cozy Entire Floor of Brownstone       4869   
4          5022   Entire Apt: Spacious Studio/Loft by central park       7192   
5          5099          Large Cozy 1 BR Apartment In Midtown East       7322   
9          5238                 Cute & Cozy Lower East Side 1 bdrm       7549   
...         ...                                                ...        ...   
48879  36480292  Gorgeous 1.5 Bdr with a private yard- Williams...     540335   
48880  36481315  The Raccoon Artist Studio in Williamsburg New ...  208514239   
48886  36483010                    Comfy 1 Bedroom in Midtown East  274311461   
48887  36483152    Garden Jewel Apartment in Williamsburg New York  208514239   
48892  36485431            Sunny Studio at Historical Neighborhood   23492952   

           host_name neighbourhood_group neighbourhood  latitude  longitude  \
1           Jennifer           Manhattan       Midtown  40.75362  -73.98377   
3        LisaRoxanne            Brooklyn  Clinton Hill  40.68514  -73.95976   
4              Laura           Manhattan   East Harlem  40.79851  -73.94399   
5              Chris           Manhattan   Murray Hill  40.74767  -73.97500   
9                Ben           Manhattan     Chinatown  40.71344  -73.99037   
...              ...                 ...           ...       ...        ...   
48879            Lee            Brooklyn  Williamsburg  40.71728  -73.94394   
48880          Melki            Brooklyn  Williamsburg  40.71232  -73.94220   
48886          Scott           Manhattan       Midtown  40.75561  -73.96723   
48887          Melki            Brooklyn  Williamsburg  40.71232  -73.94220   
48892  Ilgar & Aysel           Manhattan        Harlem  40.81475  -73.94867   

             room_type  price  minimum_nights  number_of_reviews last_review  \
1      Entire home/apt    225               1                 45  2019-05-21   
3      Entire home/apt     89               1                270  2019-07-05   
4      Entire home/apt     80              10                  9  2018-11-19   
5      Entire home/apt    200               3                 74  2019-06-22   
9      Entire home/apt    150               1                160  2019-06-09   
...                ...    ...             ...                ...         ...   
48879  Entire home/apt    120              20                  0         NaN   
48880  Entire home/apt    120               1                  0         NaN   
48886  Entire home/apt    200               6                  0         NaN   
48887  Entire home/apt    170               1                  0         NaN   
48892  Entire home/apt    115              10                  0         NaN   

       reviews_per_month  calculated_host_listings_count  availability_365  
1                   0.38                               2               355  
3                   4.64                               1               194  
4                   0.10                               1                 0  
5                   0.59                               1               129  
9                   1.33                               4               188  
...                  ...                             ...               ...  
48879                NaN                               1                22  
48880                NaN                               3               365  
48886                NaN                               1               176  
48887                NaN                               3               365  
48892                NaN                               1                27  

[25409 rows x 16 columns]

In [21]:
reviews_per_room_type_per_month = dict()
for row, col in air_bnb.iterrows():
    if col['room_type'] not in reviews_per_room_type_per_month:
        reviews_per_room_type_per_month[col['room_type']] = col['reviews_per_month']
    elif not np.isnan(col['reviews_per_month']) and not np.isnan(col['reviews_per_month']): 
        reviews_per_room_type_per_month[col['room_type']] = (col['reviews_per_month']+reviews_per_room_type_per_month[col['room_type']])/2
room_type_with_max_reviews_per_month = max(reviews_per_room_type_per_month, key = reviews_per_room_type_per_month.get)
room_type_with_max_reviews_per_month_df = air_bnb.loc[air_bnb['room_type'] == room_type_with_max_reviews_per_month]
print(f"{room_type_with_max_reviews_per_month}'s gets the most reviews per month.")
print(reviews_per_room_type_per_month)
room_type_with_max_reviews_per_month_df

Entire home/apt's gets the most reviews per month.
{'Private room': 1.04687512417263, 'Entire home/apt': 1.562564612433011, 'Shared room': 1.0952106572206446}


id                                               name    host_id  \
1          2595                              Skylit Midtown Castle       2845   
3          3831                    Cozy Entire Floor of Brownstone       4869   
4          5022   Entire Apt: Spacious Studio/Loft by central park       7192   
5          5099          Large Cozy 1 BR Apartment In Midtown East       7322   
9          5238                 Cute & Cozy Lower East Side 1 bdrm       7549   
...         ...                                                ...        ...   
48879  36480292  Gorgeous 1.5 Bdr with a private yard- Williams...     540335   
48880  36481315  The Raccoon Artist Studio in Williamsburg New ...  208514239   
48886  36483010                    Comfy 1 Bedroom in Midtown East  274311461   
48887  36483152    Garden Jewel Apartment in Williamsburg New York  208514239   
48892  36485431            Sunny Studio at Historical Neighborhood   23492952   

           host_name neighbourhood_group neighbourhood  latitude  longitude  \
1           Jennifer           Manhattan       Midtown  40.75362  -73.98377   
3        LisaRoxanne            Brooklyn  Clinton Hill  40.68514  -73.95976   
4              Laura           Manhattan   East Harlem  40.79851  -73.94399   
5              Chris           Manhattan   Murray Hill  40.74767  -73.97500   
9                Ben           Manhattan     Chinatown  40.71344  -73.99037   
...              ...                 ...           ...       ...        ...   
48879            Lee            Brooklyn  Williamsburg  40.71728  -73.94394   
48880          Melki            Brooklyn  Williamsburg  40.71232  -73.94220   
48886          Scott           Manhattan       Midtown  40.75561  -73.96723   
48887          Melki            Brooklyn  Williamsburg  40.71232  -73.94220   
48892  Ilgar & Aysel           Manhattan        Harlem  40.81475  -73.94867   

             room_type  price  minimum_nights  number_of_reviews last_review  \
1      Entire home/apt    225               1                 45  2019-05-21   
3      Entire home/apt     89               1                270  2019-07-05   
4      Entire home/apt     80              10                  9  2018-11-19   
5      Entire home/apt    200               3                 74  2019-06-22   
9      Entire home/apt    150               1                160  2019-06-09   
...                ...    ...             ...                ...         ...   
48879  Entire home/apt    120              20                  0         NaN   
48880  Entire home/apt    120               1                  0         NaN   
48886  Entire home/apt    200               6                  0         NaN   
48887  Entire home/apt    170               1                  0         NaN   
48892  Entire home/apt    115              10                  0         NaN   

       reviews_per_month  calculated_host_listings_count  availability_365  
1                   0.38                               2               355  
3                   4.64                               1               194  
4                   0.10                               1                 0  
5                   0.59                               1               129  
9                   1.33                               4               188  
...                  ...                             ...               ...  
48879                NaN                               1                22  
48880                NaN                               3               365  
48886                NaN                               1               176  
48887                NaN                               3               365  
48892                NaN                               1                27  

[25409 rows x 16 columns]

# Final Conclusion

In this cell, write your final conclusion for each of the questions asked.

Also, if you uncovered some more details that were not asked above, please discribe them here.

-- Add your conclusion --

1. Maya is busiest based on the total number of reviews and Louann is the busiest based on the number of reviews they get per month.

2. There are 5 neighborhood groups available and Staten Island is the neighborhood group that shows up the most.

3. Private rooms are not the most popular in Manhattan, Shared room's are the most popular in Manhattan.

4. Maya is busiest based on the total number of reviews and Louann is the busiest based on the number of reviews they get per month.

5. Staten Island has the highest average price at $114.81.

6. Manhattan has the highest total price.

7. Kathrine, Erin, Jelena, Matt, and Amy have the highest total price.

8. There are 162 hosts with no availability with a review count of 100 or more (see 'Out[32]').

9. Sonder (NYC) has the highest total price and is located in Manhattan.

10. Danielle from Queens last received a review on July 